packages/utilities/psammead-storybook-helpers/src/withServicesKnob.jsx

```diff
@@ -2,20 +2,41 @@ import React from 'react';
 import { Helmet } from 'react-helmet';
 import { select } from '@storybook/addon-knobs';
 import * as scripts from '@bbc/gel-foundations/scripts';
-import LANGUAGE_VARIANTS from './text-variants';
+import EXAMPLES from './text-variants';
+
+const DEFAULT_SERVICE = 'news';
+const SERVICES_LIST = Object.keys(EXAMPLES);
 
 export default ({
-  defaultService = 'news',
-  services = Object.keys(LANGUAGE_VARIANTS),
+  defaultService = DEFAULT_SERVICE,
+  services = SERVICES_LIST,
 } = {}) => storyFn => {
-  const includesService = service => services.includes(service);
-  const service = select(
+  const getServiceAndVariant = selectedItem =>
+    EXAMPLES[selectedItem].variant
+      ? {
+          service: EXAMPLES[selectedItem].service,
+          variant: EXAMPLES[selectedItem].variant,
+        }
+      : { service: selectedItem };
+
+  const includesSelectedService = selectedItem =>
+    services.includes(selectedItem);
+
+  const selectedItem = select(
     'Select a service',
-    services.filter(includesService),
+    services.filter(includesSelectedService),
     defaultService,
   );
-  const { text, script, locale, dir = 'ltr' } = LANGUAGE_VARIANTS[service];
-  const storyProps = { text, script: scripts[script], locale, dir, service };
+
+  const { text, script, locale, dir = 'ltr' } = EXAMPLES[selectedItem];
+
+  const storyProps = {
+    text,
+    script: scripts[script],
+    locale,
+    dir,
+    ...getServiceAndVariant(selectedItem),
```

Comment:
what about something like

```js
  service: getService(),
  ...(hasVariant && { variant: getVariant() },
}
```

Reply:
👌 yeah that much easier to read.

Reply:
ah slight issue with this is that, depending on if the service has a variant or not. The data structure is slightly different. Our non variant service, we get the service value from the key value but in the instance of a variant we would get it from the value of service within the object.

i.e 
```
news: {
 text: 'this is text'
},
serbianCyr: {
service: 'serbian',
variant: 'cyr',
text: 'this is text'
}
```

Reply:
Would this be a good implementation for it:
variant will just be passed into storyProps as null for non variant services
```
  const hasVariant = selectedItem =>
    typeof EXAMPLES_TEXT[selectedItem].variant === 'string';

  const getService = selectedItem =>
    hasVariant(selectedItem)
      ? EXAMPLES_TEXT[selectedItem].service
      : selectedItem;

  const getVariant = selectedItem =>
    hasVariant(selectedItem) ? EXAMPLES_TEXT[selectedItem].variant : null;

  const storyProps = {
    text,
    script: scripts[script],
    locale,
    dir,
    service: getService(selectedItem),
    variant: getVariant(selectedItem),
  };

```

Reply:
Looks good but a few things I see...

- No need for typeof stuff in hasVariant. Do `Boolean(EXAMPLES_TEXT[selectedItem].variant) instead`
- I'd make `hasVariant` a boolean argument in `getService`
- I'd make `getVariant` simpler by having the conditional logic outside of the function and just returning `EXAMPLES_TEXT[selectedItem].variant` like in my initial comment

also why did you go for `EXAMPLES_TEXT` and not `TEXT_EXAMPLES`?

Reply:
> also why did you go for EXAMPLES_TEXT and not TEXT_EXAMPLES?

i miss read the initial comment regarding that, will change to `TEXT_EXAMPLES` now 👍 

and thanks for the other feedback 💯